### PR TITLE
Improve documentation for custom implemented rules

### LINKED ISSUE #1429: Documentation is not clear on how to use pluggy to enable custom rules
## Issue
The documentation doesn't mention how pluggy library works, so people who are not familiar with it will face difficulty in running the custom implemented rules.

## Suggested Improvement

It's better to mention in the [documentation](https://docs.sqlfluff.com/en/stable/developingplugins.html#few-things-to-note-about-plugins) that we should run `pip install -e .` inside the plugin folder to include the custom rules in linting.

### PR DESCRIPTION
**Pull request by Aditya-Tripuraneni**
### Brief summary of the change made
Previously the documentation did not mention how the _pluggy_ library works. People not familiar with the library will face issues when running custom implemented rules.  Added the documentation that supports how to include custom rules in linting. 
Fixes  #1429

### Are there any other side effects of this change that we should be aware of?
No. 

### Pull Request checklist
- [x] Please confirm you have completed any of the necessary steps below.

- Included test cases to demonstrate any code changes, which may be one or more of the following:
  - `.yml` rule test cases in `test/fixtures/rules/std_rule_cases`.
  - `.sql`/`.yml` parser test cases in `test/fixtures/dialects` (note YML files can be auto generated with `tox -e generate-fixture-yml`).
  - Full autofix test cases in `test/fixtures/linter/autofix`.
  - Other.
- Added appropriate documentation for the change.
- Created GitHub issues for any relevant followup/future enhancements if appropriate.

### CODE REVIEW

File: docs/source/developingplugins.rst [hunk 1]
```diff
@@ -41,6 +41,8 @@ and is meant to categorize rules; you could use the
 letter 'S' to denote rules that enforce security checks
 for example.
 
+An important thing to note when running custom implemented rules. Run ``pip install - e`` inside the plugin folder so  custom rules in linting are included.
```
Comment (barrywhart) [inline docs/source/developingplugins.rst:44]:
```suggestion
An important thing to note when running custom implemented rules:  Run ``pip install - e .`` inside
the plugin folder so custom rules in linting are included.
```

Comment (barrywhart) [reply docs/source/developingplugins.rst:44]: Above, I suggested splitting the long line. I also added `.` to the `pip install` command -- I think that's required.

Comment (Aditya-Tripuraneni) [reply docs/source/developingplugins.rst:44]: Splitting the line seems fine, however I don't think the `.` was necessary, maybe a `,` outside of the code block? It just may confuse users with actually adding the period to the command, when in reality we don't want to add that period to the command. Good suggestion on the line split!

Comment (tunetheweb) [reply docs/source/developingplugins.rst]: I'm pretty sure it is needed, and also that you can't have a space between the dash and the `e`. Will make a suggestion on the latest PR.

File: docs/source/developingplugins.rst [hunk 2]
```diff
@@ -41,6 +41,9 @@ and is meant to categorize rules; you could use the
 letter 'S' to denote rules that enforce security checks
 for example.
 
+An important thing to note when running custom implemented rules: 
+Run ``pip install - e``, inside the plugin folder so  custom rules in linting are included.
```
Comment (tunetheweb) [inline docs/source/developingplugins.rst:45]:
```suggestion
Run ``pip install -e .``, inside the plugin folder so  custom rules in linting are included.
```

File: docs/source/developingplugins.rst [hunk 3]
```diff
@@ -41,6 +41,10 @@ and is meant to categorize rules; you could use the
 letter 'S' to denote rules that enforce security checks
 for example.
 
+An important thing to note when running custom implemented rules:
```
Comment (tunetheweb) [inline docs/source/developingplugins.rst:44]:
```suggestion
An important thing to note when running custom implemented rules:
```

**Pull request merged**